src/test/java/wolox/training/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,140 @@
+package wolox.training.controllers;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+
+
+@RunWith(SpringRunner.class)
+@AutoConfigureMockMvc
+@WebMvcTest(UserController.class)
+@AutoConfigureTestDatabase(replace = Replace.NONE)
+public class UserControllerTest extends BaseControllerTest {
+
+  @MockBean
+  private UserRepository mockUserRepository;
+  @MockBean
+  private BookRepository mockBookRepository;
+
+
+  private User user = new User();
+  private Book book = new Book();
+  private String userAsJson = "{\"id\":1,\"username\":\"Username\",\"name\":\"El Nombre\",\"birthDate\":\"2020-04-04\",\"books\":[{\"id\":0,\"author\":null,\"image\":null,\"title\":null,\"subtitle\":null,\"publisher\":null,\"year\":null,\"pages\":0,\"isbn\":null,\"users\":[]}]}";
```

Comment:
Again, this ugly thing right here.

Reply:
Check some previous comment, I think it can help you with this